geode-core/src/main/java/org/apache/geode/internal/net/NioSslEngine.java

```diff
@@ -308,18 +308,17 @@ void expandPeerAppData(ByteBuffer wrappedBuffer) {
     }
   }
 
-  /*
-   * NioSslEngine doesn't need to ensure capacity in network buffers because they
-   * are fixed in size by the SslContext. The size recommended by the context is
-   * big enough for the SslEngine to do its work.
-   */
   @Override
   public ByteBuffer ensureWrappedCapacity(int amount, ByteBuffer wrappedBuffer,
       Buffers.BufferType bufferType, DMStats stats) {
-    if (wrappedBuffer == null) {
-      wrappedBuffer = Buffers.acquireBuffer(bufferType, amount, stats);
+    ByteBuffer buffer = wrappedBuffer;
+    int requiredSize = engine.getSession().getPacketBufferSize();
+    if (buffer == null) {
+      buffer = Buffers.acquireBuffer(bufferType, requiredSize, stats);
+    } else if (buffer.capacity() < requiredSize) {
```

Comment:
Is it possible for the passed in capacity to be greater than the required size?  What happens in that case?  Is that allowable by the SSL library we are using?

Reply:
The buffer size for SSL decryption is determined by the SSLSession.  Creating a buffer bigger than that is useless.  In that case we create a decrypt buffer that is big enough to hold the entire message (which is the same size as the tcp/ip socket's input buffer).For non-SSL communications though the encrypted buffer is the same as the decrypted buffer (since no decryption is necessary).  In that case we need the buffer to be big enough to hold the entire message.